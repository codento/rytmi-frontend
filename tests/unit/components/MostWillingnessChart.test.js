import { shallowMount, createLocalVue } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import { merge } from 'lodash'
import MostWillingessChart from '@/components/Dashboard/MostWillingnessChart'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

function createWrapper (overrideMountingOptions) {
  const defaultMountingOptions = {
    localVue
  }
  const mergedMountingOptions = merge(defaultMountingOptions, overrideMountingOptions)
  return shallowMount(MostWillingessChart, mergedMountingOptions)
}

const mockSkillInfo = {
  1: {
    name: 'SkillOne',
    proficiencies: [2],
    willingness: [4],
    proficiencyAverage: 2,
    willingnessAverage: 4
  },
  2: {
    name: 'SkillTwo',
    proficiencies: [2, 5, 5],
    willingness: [2, 5, 5],
    proficiencyAverage: 4,
    willingnessAverage: 4
  }
}

describe('MostWillingnessChart.vue', () => {
  it('Should render chart with text telling information about the most wanted skill', () => {
    const propsData = {
      skillInfo: mockSkillInfo
    }
    const expected = { name: 'SkillTwo', currentKnowledgeAverage: 4, numberOfPeople: 2 }
    const wrapper = createWrapper({ propsData })
    expect(wrapper.vm.mostWillingnessToWorkWith).toEqual(expected)
  })

  it('Template is correct', () => {
    const propsData = {
      skillInfo: mockSkillInfo
    }
    const wrapper = createWrapper({ propsData })
    expect(wrapper.element).toMatchSnapshot()
  })
})
