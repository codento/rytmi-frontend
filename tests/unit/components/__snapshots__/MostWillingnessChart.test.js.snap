// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostWillingnessChart.vue Template is correct 1`] = `
<chart-card-stub>
  <template>
    <h2>
      Preferred skills
    </h2>
  </template>
   
  <b-col-stub
    tag="div"
  >
    <p
      class="lead px-2"
    >
      
      Employees are most enthusiastic about
      
      <strong>
        SkillTwo
      </strong>
      . 
      <br />
      
      There are 
      <strong>
        2
      </strong>
      
      who would prefer working with the tech
      and our current skill level average is
      
      <strong>
        4
      </strong>
    </p>
  </b-col-stub>
</chart-card-stub>
`;
